### ISSUE TITLE
[Bug]: Librispeech Recipe XLSR

### ISSUE THREAD

**Issue by matthewkperez** [speechbrain/speechbrain#2100]
### Describe the bug

I'm running the Librispeech/ASR/CTC recipe with an XLSR model (facebook/wav2vec2-large-xlsr-53). However, I'm having trouble getting the model to train properly. This is also the case in my personal experiments on a private dataset but I have trouble using a pretrained XLSR model in a seq2seq architecture where the encoder is the pretrained XLSR model.

### Expected behaviour

Finetune the model to librispeech

### To Reproduce

```
# ################################
# Model: wav2vec2 + DNN + CTC
# Augmentation: SpecAugment
# Authors: Sung-Lin Yeh 2021
# ################################

# Seed needs to be set at top of yaml, before objects with parameters are made
seed: 1986
__set_seed: !apply:torch.manual_seed [!ref <seed>]
output_folder: !ref results/XLSR_char/<seed>
wer_file: !ref <output_folder>/wer.txt
save_folder: !ref <output_folder>/save
train_log: !ref <output_folder>/train_log.txt

# URL for the biggest Fairseq english wav2vec2 model.
# wav2vec2_hub: facebook/wav2vec2-large-960h-lv60-self
wav2vec2_hub: facebook/wav2vec2-large-xlsr-53
wav2vec2_folder: !ref <save_folder>/wav2vec2_checkpoint

# Data files
data_folder: /y/public/data/LibriSpeech # e,g./path/to/LibriSpeech
# noise/ris dataset will automatically be downloaded
# data_folder_rirs: !ref <data_folder>
train_splits: ["train-clean-100"]
dev_splits: ["dev-clean"]
test_splits: ["test-clean"]
skip_prep: False
ckpt_interval_minutes: 25 # save checkpoint every N min
train_csv: !ref <output_folder>/train.csv
valid_csv: !ref <output_folder>/dev-clean.csv
test_csv:
   - !ref <output_folder>/test-clean.csv

# Training parameters
number_of_epochs: 10
lr: 0.9
lr_wav2vec: 0.0001
sorting: ascending
auto_mix_prec: False
sample_rate: 16000


batch_size: 8
test_batch_size: 8

# Dataloader options
train_dataloader_opts:
   batch_size: !ref <batch_size>

valid_dataloader_opts:
   batch_size: !ref <batch_size>

test_dataloader_opts:
   batch_size: !ref <test_batch_size>

# Model parameters
activation: !name:torch.nn.LeakyReLU
dnn_layers: 2
dnn_neurons: 1024
freeze_wav2vec: True

# Outputs
output_neurons: 29  # BPE size, index(blank/eos/bos) = 0 (31 is og)

# Decoding parameters
blank_index: 0

#
# Functions and classes
#
epoch_counter: !new:speechbrain.utils.epoch_loop.EpochCounter
   limit: !ref <number_of_epochs>

augmentation: !new:speechbrain.lobes.augment.TimeDomainSpecAugment
   sample_rate: !ref <sample_rate>
   speeds: [95, 100, 105]

enc: !new:speechbrain.lobes.models.VanillaNN.VanillaNN
   input_shape: [null, null, 1024]
   activation: !ref <activation>
   dnn_blocks: !ref <dnn_layers>
   dnn_neurons: !ref <dnn_neurons>

wav2vec2: !new:speechbrain.lobes.models.huggingface_wav2vec.HuggingFaceWav2Vec2
   source: !ref <wav2vec2_hub>
   output_norm: True
   freeze: !ref <freeze_wav2vec>
   save_path: !ref <wav2vec2_folder>

#####
# Uncomment this block if you prefer to use a Fairseq pretrained model instead
# of a HuggingFace one. Here, we provide an URL that is obtained from the
# Fairseq github for the multilingual XLSR.
#
#wav2vec2_url: https://dl.fbaipublicfiles.com/fairseq/wav2vec/wav2vec_vox_960h_pl.pt
#wav2vec2: !new:speechbrain.lobes.models.fairseq_wav2vec.FairseqWav2Vec2
#    pretrained_path: !ref <wav2vec2_url>
#    output_norm: True
#    freeze: False
#    save_path: !ref <save_folder>/wav2vec2_checkpoint/model.pt

ctc_lin: !new:speechbrain.nnet.linear.Linear
   input_size: !ref <dnn_neurons>
   n_neurons: !ref <output_neurons>

log_softmax: !new:speechbrain.nnet.activations.Softmax
   apply_log: True

ctc_cost: !name:speechbrain.nnet.losses.ctc_loss
   blank_index: !ref <blank_index>

modules:
   wav2vec2: !ref <wav2vec2>
   enc: !ref <enc>
   ctc_lin: !ref <ctc_lin>

model: !new:torch.nn.ModuleList
   - [!ref <enc>, !ref <ctc_lin>]

model_opt_class: !name:torch.optim.Adadelta
   lr: !ref <lr>
   rho: 0.95
   eps: 1.e-8

wav2vec_opt_class: !name:torch.optim.Adam
   lr: !ref <lr_wav2vec>

lr_annealing_model: !new:speechbrain.nnet.schedulers.NewBobScheduler
   initial_value: !ref <lr>
   improvement_threshold: 0.0025
   annealing_factor: 0.8
   patient: 0

lr_annealing_wav2vec: !new:speechbrain.nnet.schedulers.NewBobScheduler
   initial_value: !ref <lr_wav2vec>
   improvement_threshold: 0.0025
   annealing_factor: 0.9
   patient: 0

label_encoder: !new:speechbrain.dataio.encoder.CTCTextEncoder

checkpointer: !new:speechbrain.utils.checkpoints.Checkpointer
   checkpoints_dir: !ref <save_folder>
   recoverables:
      wav2vec2: !ref <wav2vec2>
      model: !ref <model>
      scheduler_model: !ref <lr_annealing_model>
      scheduler_wav2vec: !ref <lr_annealing_wav2vec>
      counter: !ref <epoch_counter>
      tokenizer: !ref <label_encoder>

train_logger: !new:speechbrain.utils.train_logger.FileTrainLogger
   save_file: !ref <train_log>

error_rate_computer: !name:speechbrain.utils.metric_stats.ErrorRateStats

cer_computer: !name:speechbrain.utils.metric_stats.ErrorRateStats
   split_tokens: True
```

Train_log.txt
```
epoch: 1, lr_model: 9.00e-01, lr_wav2vec: 1.00e-04 - train loss: 2.87 - valid loss: 3.00, valid CER: 98.58, valid WER: 100.00
epoch: 2, lr_model: 9.00e-01, lr_wav2vec: 1.00e-04 - train loss: 2.72 - valid loss: 2.96, valid CER: 97.98, valid WER: 100.00
epoch: 3, lr_model: 9.00e-01, lr_wav2vec: 1.00e-04 - train loss: 2.67 - valid loss: 3.01, valid CER: 97.92, valid WER: 100.00
epoch: 4, lr_model: 7.20e-01, lr_wav2vec: 9.00e-05 - train loss: 2.65 - valid loss: 2.95, valid CER: 97.32, valid WER: 100.00
epoch: 5, lr_model: 7.20e-01, lr_wav2vec: 9.00e-05 - train loss: 2.61 - valid loss: 2.94, valid CER: 97.12, valid WER: 100.00
epoch: 6, lr_model: 7.20e-01, lr_wav2vec: 9.00e-05 - train loss: 2.60 - valid loss: 2.94, valid CER: 97.36, valid WER: 100.00
epoch: 7, lr_model: 5.76e-01, lr_wav2vec: 8.10e-05 - train loss: 2.58 - valid loss: 2.94, valid CER: 97.20, valid WER: 100.00
epoch: 8, lr_model: 4.61e-01, lr_wav2vec: 7.29e-05 - train loss: 2.59 - valid loss: 2.93, valid CER: 97.13, valid WER: 100.00
epoch: 9, lr_model: 3.69e-01, lr_wav2vec: 6.56e-05 - train loss: 2.55 - valid loss: 2.91, valid CER: 96.99, valid WER: 100.00
epoch: 10, lr_model: 3.69e-01, lr_wav2vec: 6.56e-05 - train loss: 2.55 - valid loss: 2.94, valid CER: 97.02, valid WER: 100.00
```

**Comment by matthewkperez** [comment]
I saw that thread and tried all of the suggestions but I'm still unable to achieve convergence for XLSR. 
I've tried both facebook/wav2vec2-large-xlsr-53 and facebook/wav2vec2-xls-r-300m with many different seeds/learning rate params. Are there any other recommendations you have for handling wav2vec2's instability? I'm very surprised to not see more issues raised if it is due to seed initialization.

**Comment by Adel-Moumen** [comment]
Hey @TParcollet and @salah-zaiem, could you please share your advices regarding training such models? Thanks!

**Comment by matthewkperez** [comment]
Some added information. when using other's FT versions (jonatasgrosman/wav2vec2-large-xlsr-53-english) I am able to achieve convergence on a english dataset (shown below). However, when trying to FT the model myself to english I am get convergence issues. Also due to my hardware constraints I am only able to run a frozen encoder architecture. However, I have tried unfreezing the last transformer layer in an attempt to FT at least part of the SSL model but still I get convergence issues in this setup too.

```
epoch: 1, lr_model: 9.00e-01, lr_wav2vec: 1.00e-04 - train loss: 4.03 - valid loss: 1.24, valid CER: 24.63, valid WER: 44.83
epoch: 2, lr_model: 9.00e-01, lr_wav2vec: 1.00e-04 - train loss: 1.89 - valid loss: 1.19, valid CER: 23.83, valid WER: 43.18
epoch: 3, lr_model: 9.00e-01, lr_wav2vec: 1.00e-04 - train loss: 1.88 - valid loss: 1.18, valid CER: 23.69, valid WER: 43.16
epoch: 4, lr_model: 9.00e-01, lr_wav2vec: 1.00e-04 - train loss: 1.87 - valid loss: 1.20, valid CER: 23.79, valid WER: 43.35
epoch: 5, lr_model: 9.00e-01, lr_wav2vec: 1.00e-04 - train loss: 1.85 - valid loss: 1.21, valid CER: 23.18, valid WER: 43.61
epoch: 6, lr_model: 7.20e-01, lr_wav2vec: 9.00e-05 - train loss: 1.74 - valid loss: 1.15, valid CER: 22.81, valid WER: 42.62
epoch: 7, lr_model: 7.20e-01, lr_wav2vec: 9.00e-05 - train loss: 1.73 - valid loss: 1.12, valid CER: 22.89, valid WER: 41.84
epoch: 8, lr_model: 7.20e-01, lr_wav2vec: 9.00e-05 - train loss: 1.75 - valid loss: 1.11, valid CER: 22.68, valid WER: 41.77
epoch: 9, lr_model: 7.20e-01, lr_wav2vec: 9.00e-05 - train loss: 1.73 - valid loss: 1.14, valid CER: 22.64, valid WER: 42.07
epoch: 10, lr_model: 7.20e-01, lr_wav2vec: 9.00e-05 - train loss: 1.72 - valid loss: 1.11, valid CER: 22.67, valid WER: 41.59
```
